### DISCUSSION TITLE
Obtenção de certificado digital para ambiente de homologação

### DISCUSSION THREAD

**Discussion by brunoffreire** [wmixvideo/nfe#889] [Geral]
Boa tarde.

Para desenvolver um sistema usando esta lib java e testar a emissão de NF em ambiente de homologação, é necessário um certificado digital verdadeiro? 

É possível desenvolver o software e testar com um certificado auto-assinado... ou obter um certificado de testes?

Alguém sabe orientar sobre esse tópico?

Obrigado.

**Comment by marcoslombardi** [comment]
Boa tarde!

Para emissão de nota fiscal eletrônica, é necessário possuir um certificado digital emitido por uma Autoridade Certificadora (AC) credenciada pela Infraestrutura de Chaves Públicas Brasileira (ICP-Brasil). Esse certificado digital deve ser do tipo A1 ou A3, e estar vinculado ao CNPJ da empresa que irá emitir as notas fiscais eletrônicas.

Tanto em homologação e em produção precisa do certificado para envio de lote. Recomendo começar pelo certificado do tipo A1 que é mais simples sua implementação.

Na classe NFeConfig configura qual ambiente quer acessar sobrescrevendo o método abaixo:

`public DFAmbiente getAmbiente() {
		return DFAmbiente.HOMOLOGACAO;
}`

**Comment by brunoffreire** [comment]
Entendido. Obrigado pela resposta.

Outra pergunta. Se o certificado digital for de uma empresa sem inscrição estadual (por que só presta serviços, não paga ICMS)... Esse certificado pode ser usado para teste de emissão de nota de venda de produtos no ambiente de homologação da SEFAZ?

Também li na internet que poderia ser usado e-cpf no ambiente de homologação da SEFAZ. Alguém sabe?

**Reply by marcoslombardi** [reply]
Para cadastrar no ambiente de emissão de nfe na SEFAZ é necessário que o CNPJ vinculado ao certificado digital tenha uma IE válida. As notas fiscais eletrônicas são referentes a produtos e não serviços. É possível emitir nfe por CPF sim, produtor rural por exemplo. Uma contabilidade "mais digital" pode te auxiliar na emissão certificado digital e cadastro na SEFAZ do seu estado.
